src/root/components/OperationalUpdatePdfDocument/index.tsx

```diff
@@ -0,0 +1,874 @@
+          <Text style={pdfStyles.piBorderCell}>
+            {data.indicator}
+          </Text>
+        </View>
+      </View>
+      <View style={pdfStyles.piRow}>
+        <View style={pdfStyles.piIconCell} />
+        <View style={pdfStyles.piHeaderCell}>
+          <Text>
+            {strings.drefExportPriorityActions}
+          </Text>
+        </View>
+        <View style={pdfStyles.piContentCell}>
+          <Text style={pdfStyles.piBorderCell}>
+            {data.description}
+          </Text>
+        </View>
+      </View>
+    </View>
+  );
+}
+
+function formatBoolean(value: boolean | undefined | null) {
+  if (value === true) {
+    return 'Yes';
+  }
+
+  if (value === false) {
+    return 'No';
+  }
+
+  return '-';
+}
+
+function formatNumber(value: number | undefined | null, prefix?: string): string {
+  const defaultValue = '-';
+
+  if (isValidNumber(value)) {
+    const formattedNumber = addSeparator(value) ?? defaultValue;
+
+    if (prefix) {
+      return `${prefix}${formattedNumber}`;
+    }
+
+    return formattedNumber;
+  }
+
+  return defaultValue;
+}
+
+function TextOutput(props: {
+  label: string;
+  value?: string;
+  columns?: '1/3' | '2/3' | '3/3' | '1/2';
+}) {
+  const {
+    label,
+    value,
+    columns = '1/3',
+  } = props;
+
+  const styleMap = {
+    '1/3': pdfStyles.oneByThree,
+    '2/3': pdfStyles.twoByThree,
+    '3/3': pdfStyles.threeByThree,
+    '1/2': pdfStyles.oneByTwo,
+  };
+
+  return (
+    <View
+      style={[
+        pdfStyles.textOutput,
+        styleMap[columns] ?? pdfStyles.oneByThree,
+      ]}
+    >
+      <Text style={pdfStyles.textOutputLabel}>
+        {label}
+      </Text>
+      <Text style={pdfStyles.textOutputValue}>
+        {value}
+      </Text>
+    </View>
+  );
+}
+
+interface DrefOptions {
+  disaster_category: NumericKeyValuePair[];
+  national_society_actions: StringKeyValuePair[];
+  needs_identified: StringKeyValuePair[];
+  planned_interventions: StringKeyValuePair[];
+  status: NumericKeyValuePair[];
+  type_of_onset: NumericKeyValuePair[];
+  users: {
+    id: number;
+    first_name: string;
+    last_name: string;
+    email: string;
+    username: string;
+  }[];
+}
+
+interface Props {
+  operationalUpdateResponse: DrefOperationalUpdateApiFields;
+  drefOptions: DrefOptions;
+  // onLoad?: (params: { blob?: Blob }) => void;
+  strings: Strings;
+}
+
+function OperationalUpdatePdfDocument(props: Props) {
+  const {
+    operationalUpdateResponse,
+    drefOptions,
+    strings,
+  } = props;
+
+  const getMaps = () => {
+    if (!drefOptions) {
+      return [
+        {},
+        {},
+        {},
+      ];
+    }
+
+    return [
+      listToMap(drefOptions.planned_interventions, d => d.key, d => d.value),
+      listToMap(drefOptions.needs_identified, d => d.key, d => d.value),
+      listToMap(drefOptions.national_society_actions, d => d.key, d => d.value),
+    ];
+  };
+  const [
+    piMap,
+    niMap,
+    nsaMap,
+  ] = getMaps();
```

Comment:
Why not use useMemo?

Reply:
Because the parent component uses the useCallback hooks.